components/script/dom/bindings/refcounted.rs

```diff
@@ -142,10 +142,7 @@ impl TrustedPromise {
         let this = self;
         task!(resolve_promise: move || {
             debug!("Resolving promise.");
-            let this = this.root();
-            let cx = this.global().get_cx();
-            let _ac = JSAutoCompartment::new(cx, this.reflector().get_jsobject().get());
```

Comment:
@jdm, do you know why did we need `JSAutoCompartment` here? Compartments are per global, I'm assuming we can't guarantee that the correct global is on scope when we execute the task?